sdk/metric/internal/aggregate/aggregate.go

```diff
@@ -0,0 +1,133 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregate // import "go.opentelemetry.io/otel/sdk/metric/internal/aggregate"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// Input receives measurements to be aggregated.
+type Input[N int64 | float64] func(context.Context, N, attribute.Set)
+
+var bgCtx = context.Background()
+
+// Async receives asynchronous measurements that do not have a context
+// associated with them.
+func (f Input[N]) Async(v N, a attribute.Set) { f(bgCtx, v, a) }
+
```

Comment:
This is only used in `func (o *observable[N]) observe(val N, s attribute.Set)`. How about removing it? See other comment.

```suggestion
```

Reply:
Having the `Async` function avoids having asynchronous functions from creating a background context at the call site and has the added benefit that all function calls will use the same instance of `context.Background` instead of creating a new one each time they want to call it.

Reply:
`context.Background()` is also the same instance

See https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/context/context.go;l=205-216

I do not believe it would ever change 😉